concrete_types.hh

```diff
@@ -15,6 +15,7 @@
 #include "types/map.hh"
 #include "types/set.hh"
 #include "types/tuple.hh"
+#include "types/vector.hh"
```

Comment:
Changed the commit message accordingly.

Although I've left the from_string implementation here so far, because I'm not sure if it's needed at all and might be removed (see the discussion [below](https://github.com/zpp-2024-vector-search/scylladb/pull/5#discussion_r1898372571)).